nflow-engine/src/main/java/io/nflow/engine/workflow/curated/CronWorkflow.java

```diff
@@ -0,0 +1,116 @@
+import io.nflow.engine.workflow.definition.StateExecution;
+import io.nflow.engine.workflow.definition.StateVar;
+import io.nflow.engine.workflow.definition.WorkflowDefinition;
+import io.nflow.engine.workflow.definition.WorkflowSettings;
+import io.nflow.engine.workflow.definition.WorkflowSettings.Builder;
+import io.nflow.engine.workflow.definition.WorkflowState;
+import io.nflow.engine.workflow.definition.WorkflowStateType;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.springframework.scheduling.support.CronSequenceGenerator;
+
+import static io.nflow.engine.workflow.curated.CronWorkflow.State.doWork;
+import static io.nflow.engine.workflow.curated.CronWorkflow.State.failed;
+import static io.nflow.engine.workflow.curated.CronWorkflow.State.handleFailure;
+import static io.nflow.engine.workflow.curated.CronWorkflow.State.schedule;
+import static io.nflow.engine.workflow.definition.NextAction.moveToState;
+import static io.nflow.engine.workflow.definition.NextAction.moveToStateAfter;
+import static io.nflow.engine.workflow.definition.WorkflowStateType.manual;
+import static io.nflow.engine.workflow.definition.WorkflowStateType.normal;
+import static io.nflow.engine.workflow.definition.WorkflowStateType.start;
+import static org.joda.time.Instant.now;
+import static org.joda.time.Period.weeks;
+import static org.slf4j.LoggerFactory.getLogger;
+
+/**
+ * Cron workflow executor that periodically wakes up.
+ */
+public abstract class CronWorkflow extends WorkflowDefinition<State> {
+  private static final Logger logger = getLogger(CronWorkflow.class);
+
+  /**
+   * Name of the state variable that stores schedule in cron syntax.
+   */
+  public static final String VAR_SCHEDULE = "cron";
+
+  public enum State implements WorkflowState {
+    schedule(start), doWork(normal), handleFailure(normal), failed(manual);
+
+    private WorkflowStateType type;
+
+    State(WorkflowStateType type) {
+      this.type = type;
+    }
+
+    @Override
+    public WorkflowStateType getType() {
+      return type;
+    }
+
+    @Override
+    public String getDescription() {
+      return name();
+    }
+  }
+
+  /**
+   * Extend cron workflow definition with customized workflow settings.
+   * Extending class must implement the 'doWork` step method.
+   *
+   * @param type The type of the workflow.
+   * @param settings The workflow settings.
+   */
+  protected CronWorkflow(String type, WorkflowSettings settings) {
+    super(type, schedule, handleFailure, settings);
+    permit(schedule, doWork);
+    permit(doWork, schedule);
+    permit(handleFailure, schedule, failed);
+  }
+
+  /**
+   * Extend cron workflow definition.
+   * Extending class must implement the 'doWork` step method.
+   *
+   * @param type The type of the workflow.
+   */
+  protected CronWorkflow(String type) {
+    this(type, new Builder().setHistoryDeletableAfter(weeks(1)).build());
+  }
+
+  /**
+   * Calculates the next activation time.
```

Comment:
Default impl calcs next activation time based on cron state variable. Override for custom scheduling.